### PR TITLE
feat: Adds ability to use styles for pre element

### LINKED ISSUE #91: Add styles support for a `<pre>` tag.
Right now it's not possible to add styles to the `<pre>` tag element.

### PR DESCRIPTION
**Pull request by butschster**
We can try to add styles for every line inside pre element
```html
<pre class="bg-blue text-red font-bold">
    <h1>Introduction</h1>

    <div>The body of your message.</div>

    Thanks,
    Laravel

    © 2021 Laravel. All rights reserved.
</pre>
```

There is a problem with line width. If I didn't pay attention to the line width, I got something like this

image

Then I decided to get width for the longest line in a html and apply `str_pad` function to every line. 

image

fixed #91

### CODE REVIEW

**Comment by xiCO2k** [comment]
Just saw, the solution looks good to me.

Found this 2 cases:

* By default it looks like it have an extra like break on start:

image

* And if you add a `my-2` style it adds it on each line.

Besides that, you are killing it 🥇 .

Thanks again @butschster.

**Comment by butschster** [comment]
> Just saw, the solution looks good to me.
> 
> Found this 2 cases:
> 
> * By default it looks like it have an extra like break on start:
> 
> image
> 
> * And if you add a `my-2` style it adds it on each line.
> 
> Besides that, you are killing it 🥇 .
> 
> Thanks again @butschster.

Oh, after style inheritance we can try to replace `my-...` with `my-0`

**Comment by butschster** [comment]
And I also noticed a problem with `italic`. Try this

```html
<pre class="bg-blue text-red font-bold italic">
    <h1>Introduction</h1>

    <div>The body of your message.</div>

    Thanks,
    Laravel

    © 2021 Laravel. All rights reserved.
</pre>
```

**Comment by xiCO2k** [comment]
Looks good to me:

image

**Comment by butschster** [comment]
> Looks good to me:
> 
> image

Hmmm))) ok...

File: tests/pre.php [hunk 1]
```diff
@@ -1,18 +1,27 @@
...
+
+    $html = parse("<pre>$content</pre>");
 
-    expect($html)->toBe('
+    expect($html)->toBe(
+        "\n".implode("\n", array_map(
+            fn (string $line) => str_pad($line, 44),
+            explode("\n", $content)
+        ))
+    );
```
Comment (xiCO2k) [inline tests/pre.php:20]: Can you change the output of this to the output text? Thanks

**Review by xiCO2k — CHANGES_REQUESTED** [review]

File: tests/pre.php (hunk 1 above)
Comment (butschster) [reply tests/pre.php:20]: There is a problem to use plain text in this test, cause PHPStorm removes all extra spaces when someone opens this file.

Comment (xiCO2k) [reply tests/pre.php:20]: Humm, maybe string concat for each line? Thanks

Comment (butschster) [reply tests/pre.php:20]: What about something like this?

```php
    expect($html)->toBe("
    <h1>Introduction</h1>".str_repeat(' ', 19)."
".str_repeat(' ', 44)."
    <div>The body of your message.</div>".str_repeat(' ', 4)."
".str_repeat(' ', 44)."
    Thanks,".str_repeat(' ', 33)."
    Laravel".str_repeat(' ', 33)."
".str_repeat(' ', 44)."
    © 2021 Laravel. All rights reserved.".str_repeat(' ', 3)
    );
```

Comment (xiCO2k) [reply tests/pre.php:20]: I think it works.

**Pull request merged**